lib/order.rb

```diff
@@ -0,0 +1,72 @@
+require_relative "customer"
+
+class Order
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+
+  def initialize(id, products, customer, fulfillment_status = :pending) # keyword argument/optional argument. Check
+    @id = id
+    @products = products # { "banana" => 1.99, "cracker" => 3.00 }
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+    if ![:pending, :paid, :processing, :shipped, :complete].include?(@fulfillment_status)
+      raise ArgumentError, "Invalid fulfillment status"
+    end
```

Comment:
Love that you added this input validation! This kind of edge-case handling will be super useful once we start getting into bigger web applications :)